tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h

```diff
@@ -59,74 +74,14 @@ class TrajoptPlanner : public BasicPlanner
-   */
-  bool solve(PlannerResponse& response,
-             const trajopt::TrajOptProbPtr& prob,
-             const trajopt::BasicTrustRegionSQPParameters& params);
-
-  /**
-   * @brief Sets up the optimizer and solves the SQP problem with a single callback and parameters passed in
-   * @param response The results of the optimization. Primary output is the optimized joint trajectory
-   * @param prob Trajopt problem to be solved
-   * @param params Optimization parameters to be used
-   * @param callback A callback function to be called on each iteration of the optimization, e.g. plotting, write to
-   * file, etc.
-   * @return true if optimization complete
-   */
-  bool solve(PlannerResponse& response,
-             const trajopt::TrajOptProbPtr& prob,
-             const trajopt::BasicTrustRegionSQPParameters& params,
-             const trajopt::Optimizer::Callback& callback);
-
-  /**
-   * @brief Sets up the optimizer and solves the SQP problem with a single callback and default parameters
-   * @param response The results of the optimization. Primary output is the optimized joint trajectory
-   * @param prob Trajopt problem to be solved
-   * @param callback A callback function to be called on each iteration of the optimization, e.g. plotting, write to
-   * file, etc.
-   * @return true if optimization complete
-   */
-  bool solve(PlannerResponse& response,
-             const trajopt::TrajOptProbPtr& prob,
-             const trajopt::Optimizer::Callback& callback);
-
-  /**
-   * @brief Sets up the optimizer and solves the SQP problem with a vector of callbacks and default parameters
-   * @param response The results of the optimization. Primary output is the optimized joint trajectory
-   * @param prob Trajopt problem to be solved
-   * @param callbacks A callback function to be called on each iteration of the optimization, e.g. plotting, write to
-   * file, etc.
-   * @return true if optimization complete
-   */
-  bool solve(PlannerResponse& response,
-             const trajopt::TrajOptProbPtr& prob,
-             const std::vector<trajopt::Optimizer::Callback>& callbacks);
-
-  /**
-   * @brief Sets up optimizer and solves a SQP problem
-   * @param response The results of the optimization. Primary output is the optimized joint trajectory
-   * @param prob Trajopt problem to be solved
-   * @param params SQP parameters for the optimization. If not given, defaults will be used
-   * @param callbacks A callback function to be called on each iteration of the optimization, e.g. plotting, write to
-   * file, etc.
+   * @param config Trajopt planner config
    * @return true if optimization complete
    */
-  bool solve(PlannerResponse& response,
-             const trajopt::TrajOptProbPtr& prob,
-             const trajopt::BasicTrustRegionSQPParameters& params,
-             const std::vector<trajopt::Optimizer::Callback>& callbacks);
+  bool solve(PlannerResponse& response, const TrajOptPlannerConfig& config);
```

Comment:
Yea, that was my original idea. So the in the `PlanningRequest` type there is a `std::string config` and std::string config-format`. The idea behind this would was to accomplish what you describe to encode the problem in say xml, json, ext. then parse this string to create the problem. In order to accomplish this for TrajOpt was to for each of the cost/constraint have a toJson where in c++ you could create a problem encode it to json and then send it as an action where the planner then parses the problem.